pkg/lib/botprotection/cloudflare_response.go

```diff
@@ -0,0 +1,62 @@
+package botprotection
+
+import (
+	"strings"
+
+	"github.com/authgear/authgear-server/pkg/util/slice"
+)
+
+// raw API response from cloudflare turnstile, which is not visible to consumer of this API client
+type CloudflareTurnstileRawResponse struct {
+	Success *bool `json:"success,omitempty"`
+
+	// non-empty if Success == false, empty if Success == true
+	ErrorCodes []CloudflareTurnstileErrorCode `json:"error-codes"`
+
+	// specific to Success == true
+	ChallengeTs string `json:"challenge_ts"` // ISO timestamp for the time the challenge was solved.
+	Hostname    string `json:"hostname"`     // hostname for which the challenge was served.
+	Action      string `json:"action"`       // customer widget identifier passed to the widget on the client side.
+	CData       string `json:"cdata"`        // customer data passed to the widget on the client side.
+}
+
+// parsed success response
+type CloudflareTurnstileSuccessResponse struct {
+	ChallengeTs string `json:"challenge_ts"` // ISO timestamp for the time the challenge was solved.
+	Hostname    string `json:"hostname"`     // hostname for which the challenge was served.
+	Action      string `json:"action"`       // customer widget identifier passed to the widget on the client side.
+	CData       string `json:"cdata"`        // customer data passed to the widget on the client side.
+}
+
+// parsed error response
+type CloudflareTurnstileErrorResponse struct {
+	ErrorCodes []CloudflareTurnstileErrorCode `json:"error-codes"`
+}
```

Comment:
I think about this again. We can reuse CloudflareTurnstileRawResponse to represent both response. When it is successfully, we return `(*CloudflareTurnstileRawResponse, nil)`, When it is an error, we return `(nil, error)` when the error COULD be `*CloudflareTurnstileRawResponse`